queue.go

```diff
@@ -106,12 +108,39 @@ type Queue struct {
 	workers      []worker
 }
 
-// NewQueue creates and returns a new Queue structure
-func NewQueue(url *URL, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
-	q := &Queue{
+// NewQueue returns a new Queue structure
+func NewQueue(url *URL, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Channel, error) {
+	q := &Channel{
 		name:         name,
 		url:          url.urlString(),
 		isConsumer:   isConsumer,
+		isExchange:   false,
```

Comment:
Do we really have to keep both `isConsumer` and `isExchange`? Is it possible to have a queue where `isConsumer` and `isExchange` are `false`?

Reply:
Yeah at the time of publishing in a queue isConsumer could be false

Reply:
@mayuree99 What would be the meaning of an object with `isConsumer` and `isExchange` are `false`? What is this if it is not a consumer (queue) and not an exchange? I don't think we need both these flags. The are just misleading